src/test/shell/integration/modify_execution_info_test.sh

```diff
@@ -98,17 +133,26 @@ EOF
 Genrule=+requires-a,CppCompile=+requires-b,CppCompile=+requires-c \
     > output 2> "$TEST_log" || fail "Expected success"
   assert_contains "ExecutionInfo: {requires-a: '', requires-z: ''}" output
-  assert_contains "ExecutionInfo: {requires-b: '', requires-c: ''}" output
+  assert_contains "ExecutionInfo: {requires-b: '', requires-c: ''" output
 
   # negative lookahead
   bazel aquery --output=text "//$pkg:all" \
    --modify_execution_info='(?!Genrule).*=+requires-a,(?!CppCompile).*=+requires-z' \
     > output 2> "$TEST_log" || fail "Expected success"
   assert_contains "ExecutionInfo: {requires-x: '', requires-z: ''}" output
-  assert_contains "ExecutionInfo: {requires-a: ''}" output
+  assert_contains "ExecutionInfo: {requires-a: ''" output
 }
 
 function test_modify_execution_info_various_types() {
+    # proto_library requires this external workspace.
+    cat >> WORKSPACE << EOF
+http_archive(
```

Comment:
Adding this rule makes the test flaky and dependent on an internet connection. Can you think of a different approach, or use an offline-compatible rule?

Reply:
This is a good point, though I took this pattern from `src/test/shell/bazel/bazel_proto_library_test.sh`.

In the latest version, I've modified this test to pull in the vendorized protobuf version from the repository.